modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/groovy/script/uses/factory/WorkflowDefinitionGroovyScriptUsesFactory.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.workflow.web.internal.groovy.script.uses.factory;
+
+import com.liferay.petra.function.transform.TransformUtil;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.GroupConstants;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.util.HttpComponentsUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.workflow.WorkflowDefinition;
+import com.liferay.portal.security.script.management.groovy.script.use.GroovyScriptUse;
+import com.liferay.portal.security.script.management.groovy.script.uses.factory.GroovyScriptUsesFactory;
+import com.liferay.portal.util.PropsValues;
+import com.liferay.portal.workflow.constants.WorkflowPortletKeys;
+import com.liferay.portal.workflow.constants.WorkflowWebKeys;
+import com.liferay.portal.workflow.manager.WorkflowDefinitionManager;
+import com.liferay.portal.workflow.portlet.tab.WorkflowPortletTabServiceTracker;
+
+import java.util.List;
+import java.util.Locale;
+
+import javax.portlet.PortletMode;
+import javax.portlet.ResourceRequest;
+import javax.portlet.WindowState;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Feliphe Marinho
+ */
+@Component(service = GroovyScriptUsesFactory.class)
+public class WorkflowDefinitionGroovyScriptUsesFactory
+	implements GroovyScriptUsesFactory {
+
+	@Override
+	public List<GroovyScriptUse> create(ResourceRequest resourceRequest)
+		throws Exception {
+
+		return TransformUtil.transform(
+			_workflowDefinitionManager.getActiveWorkflowDefinitions(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS),
+			workflowDefinition -> {
+				String content = workflowDefinition.getContentAsXML();
+
+				if (!content.contains(
+						"<script-language>groovy</script-language>")) {
+
+					return null;
+				}
```

Comment:
Hey @brianchandotcom let me know if you prefer I use the same [approach](https://github.com/brianchandotcom/liferay-portal/pull/148794/commits/24243d0d13add2429893490c39430aa38982e579) here as well